src/modules/formBuilder/components/itemEditor/pickLists/PickListOption.tsx

```diff
@@ -0,0 +1,111 @@
+import { Stack } from '@mui/material';
+import { useCallback } from 'react';
+import ControlledCheckbox from '@/modules/form/components/rhf/ControlledCheckbox';
+import ControlledTextInput from '@/modules/form/components/rhf/ControlledTextInput';
+import { FormItemControl } from '@/modules/formBuilder/components/itemEditor/types';
+import { Component } from '@/types/gqlTypes';
+
+interface PickListOptionProps {
+  control: FormItemControl;
+  index: number;
+  formItemComponent: Component;
+  setPickListInitialSelected: (code: string) => void;
+  isCodeUnique: (coe: string) => boolean;
+  code?: string;
+}
+
+const PickListOption: React.FC<PickListOptionProps> = ({
+  control,
+  code,
+  index,
+  formItemComponent,
+  isCodeUnique,
+  setPickListInitialSelected,
+}) => {
+  const onChangeInitialSelected = useCallback(
+    (value: boolean) => {
+      // Even though it's a controlled RHF checkbox, we still provide an onChange callback
+      // in order to set all the other pick list options' `initialSelected` values to false when this one is set to true.
+      if (value && code) setPickListInitialSelected(code);
+    },
+    [setPickListInitialSelected, code]
+  );
+
+  return (
+    <Stack gap={2}>
+      <ControlledTextInput
+        control={control}
+        name={`pickListOptions.${index}.code`}
+        label='Code'
+        helperText='Must be unique'
+        required={true}
+        rules={{
+          validate: (input) => isCodeUnique(input),
+        }}
+      />
+      <ControlledTextInput
+        control={control}
+        name={`pickListOptions.${index}.label`}
+        label='Label'
+        helperText='If Label is not provided, Code will be displayed'
+      />
+
+      <ControlledCheckbox
+        name={`pickListOptions.${index}.initialSelected`}
+        control={control}
+        label='Initially selected'
+        helperText='Whether this choice is selected by default'
+        onChange={onChangeInitialSelected}
```

Comment:
Same here, could watch `pickListOptions.${index}.initialSelected` and call `onChangeInitialSelected` in a useEffect. I think it is a little more explanatory, because it really is happening as a _side effect_ of the change that occurred. Supplying an `onChange` prop for these controlled components is confusing imo since they actually don't support being controlled from anything other than RHF.